src/main/java/hudson/plugins/parameterizedtrigger/BuildInfoExporterAction.java

```diff
@@ -243,40 +380,48 @@ public void buildEnvVars(AbstractBuild<?, ?> build, EnvVars env) {
    * @return
    */
   public Object readResolve() {
-    if (this.lastReference == null) {
-      this.lastReference = new BuildReference(this.buildName, this.buildNumber, Result.NOT_BUILT);
+    if (this.lastReference2 == null) {
+      this.lastReference2 = this.lastReference;
```

Comment:
Forget my last comment, the problem is I tried to testing using plain xstream without using the jenkins specific xstream configuration. What a mess, depending on which documentation one is looking it is working or not. If you have a look at the BuildInfoExporterTest you will find a testcase called "testMigrateFrom221", this test loads an old 2.21 configuartion using the jenkins internal implementation. That test case will restore the lastReference although its set to transient since jenkins has configured xstream to do so. Since lastReference is only used in readResolve and the test is green this works. Probably you just overlooked that lastReference2 is used in addBuild, the constructor, getLastReference and here in readResolve as well. If the instance is serialized through xstream only lastReference2 is stored.
Do you know whether it's possible to switch to the ConverterImpl approach? This would be by far the cleaner way to separate the underlying storage format from the implementation.